src/main/java/uk/gov/homeoffice/digital/sas/timecard/listeners/KafkaEntityListener.java

```diff
@@ -0,0 +1,25 @@
+package uk.gov.homeoffice.digital.sas.timecard.listeners;
+
+import uk.gov.homeoffice.digital.sas.timecard.enums.KafkaAction;
+import uk.gov.homeoffice.digital.sas.timecard.kafka.producers.KafkaProducerService;
+
+public abstract class KafkaEntityListener<T> {
+
+  protected KafkaProducerService<T> kafkaProducerService;
+
+  public abstract String resolveMessageKey(T resource);
+
+  protected void createProducerService(KafkaProducerService<T> kafkaProducerService) {
+    this.kafkaProducerService = kafkaProducerService;
+  }
+
+  protected void sendKafkaMessageOnCreate(T resource) {
+    kafkaProducerService.sendMessage(resolveMessageKey(resource),
+        (Class<T>) resource.getClass(), resource, KafkaAction.CREATE);
+  }
+
+  protected void sendKafkaMessageOnUpdate(T resource) {
+    kafkaProducerService.sendMessage(resolveMessageKey(resource),
+        (Class<T>) resource.getClass(), resource, KafkaAction.UPDATE);
```

Comment:
Can extract the `kafkaProducerService.sendMessage` call into a private method that takes the `resource` &  `KafkaAction`.
Cleans it up a bit and shows the only difference between the two calls is the action.
Feel free to resolve if you don't think it's worthwhile.

Reply:
Updated